src/Build/Graph/ProjectGraph.cs

```diff
@@ -598,6 +601,24 @@ private string DebuggerDisplayString()
         {
             ThrowOnEmptyTargetNames(entryProjectTargets);
 
+            // Solutions have quirky behavior when provided a target with ';' in it, eg "Clean;Build". This can happen if via the command-line the user provides something
+            // like /t:"Clean;Build". When building a project, the target named "Clean;Build" is executed (which usually doesn't exist, but could). However, for solutions
+            // the generated metaproject ends up calling the MSBuild task with the provided targets, which ends up splitting the value as if it were [ "Clean", "Build" ].
+            // Mimic this flattening behavior for consistency.
+            if (_isSolution && entryProjectTargets != null && entryProjectTargets.Count != 0)
+            {
+                List<string> newEntryTargets = new(entryProjectTargets.Count);
+                foreach (string entryTarget in entryProjectTargets)
+                {
+                    foreach (string s in ExpressionShredder.SplitSemiColonSeparatedList(entryTarget))
+                    {
+                        newEntryTargets.Add(s);
+                    }
+                }
+
+                entryProjectTargets = newEntryTargets;
+            }
```

Comment:
Why recreating the list twice? (`SplitSemiColonSeparatedList` already creates new)

```suggestion
            {
                entryProjectTargets = ExpressionShredder.SplitSemiColonSeparatedList(entryTarget);
            }
```